client/state/themes/actions/block-themes.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+import {
+	BLOCK_THEMES_FAIL,
+	BLOCK_THEMES_FETCH,
+	BLOCK_THEMES_SUCCESS,
+} from 'calypso/state/themes/action-types';
+
+import 'calypso/state/themes/init';
+
+/**
+ * Receives themes and dispatches them with a block themes success signal.
+ *
+ * @param {Array} themes array of received theme objects
+ * @returns {Function} Action thunk
+ */
+export function receiveBlockThemes( themes ) {
+	return ( dispatch ) => {
+		dispatch( { type: BLOCK_THEMES_SUCCESS, payload: themes } );
+	};
+}
+
+/**
+ * Initiates a network request for block themes.
+ * Block themes are for the Site Editor and are denoted by the 'block-templates' tag.
+ *
+ * @returns {Function} Action thunk
+ */
+export function getBlockThemes() {
```

Comment:
With the exception of `filter` value, I think this is pretty much the same as `getRecommendedThemes`. Should we abstract this further and pass the `filter` as an argument? That way both recommended and block themes could reuse the same code, and we'd reduce the amount of redux boilerplate.

Reply:
I considered that, but didn't simply due to the fact that I wanted to keep the storage for the current `recommendedThemes` to stay separate. I guess I could abstract a bit more within the store… I'll think about it a bit more.

Reply:
I guess that one drawback of doing that is that we'll be modifying existing code, and there is some risk of causing regressions for a temporary piece of functionality. On the other hand, there might be more theme types that we'd want to filter in a similar way down the road (e.g. universal, hybrid, legacy) depending on how we design these theme switching flows.

Reply:
I think I agree with @vindl here. 🤔 

The fact that we are reusing the `RecommendedThemes` component to _maybe_ show block themes instead of recommended ones, might be a hint of code smell that could obfuscate the intention of the component.

Block themes are recommended themes too, just we recommend them to some specific sites.

As long as the `recommendedThemes` are cleared appropriately when switching sites (or changing relevant site options, etc.), I'm not sure I understand the value of having a separate `blockThemes` list. 🤔

Reply:
I'm coming around on this. I initially thought about it a different way, that the current iteration of `recommendedThemes` would go away and be replace by `blockThemes` in time. I also thought that Block Themes were important enough that we would be using these selectors elsewhere, too.

But the second case is only hypothetical, and the to accomplish the first, it would still be better to do as suggested here, to just use `recommendThemes` across the board. It does add a layer of complexity, as I will need to disambiguate in the redux store, which initially seemed difficult, but I'll figure it out.